lib/async-compiler/adapters/take2-adapter.js

```diff
@@ -1,12 +1,26 @@
 import URLAdapter from './url-adapter.js';
 
+export function Take2AdapterException(message) {
+  this.message = message;
+  this.name = 'Take2AdapterException';
+}
 
-export const take2ApiHost = process.env.TAKE2_API_HOST || 'http://take2-loopback.herokuapp.com/api/v1';
-export const take2PublicKey = process.env.TAKE2_PUBLIC_KEY || 'pk_somefakekey';
+export default class extends URLAdapter {
 
+  constructor(options) {
+    super(options);
 
-export default class extends URLAdapter {
+    if(!options.take2PublicKey) {
+      throw new Take2AdapterException(`Take2 key is not defined.  Cannot create adapter.`);
+    }
+
+    if(!options.take2ApiHost) {
+      throw new Take2AdapterException(`Take2 host is not defined.  Cannot create adapter.`);
+    }
 
+    this.TAKE2_PUBLIC_KEY = options.take2PublicKey;
```

Comment:
We are currently using the SECRET_KEY for this, not the public one. Maybe it's a good time to rename it on async-compiler?